cdap-api/src/main/java/io/cdap/cdap/api/app/ApplicationSpecification.java

```diff
@@ -63,6 +63,14 @@ public interface ApplicationSpecification {
   @Nullable
   String getConfiguration();
 
+  /**
+   * @return Change summary for the update of the Application.
+   */
+  @Nullable
+  default String getChangeSummary() {
```

Comment:
I think it would be better to make a ChangeSummary class for this instead of a String. Would make it more extensible, in case we decide we want to add other information, such as the author, timestamp, or parent version

Reply:
We store only change summary in app-spec, other information (author, creation time) is written directly to the table - since in the future there might be a need to query directly off either of those values. However, we can return this as an object in `ApplicationDetail`?